nexus/src/app/sagas/snapshot_create.rs

```diff
@@ -181,6 +206,39 @@ impl NexusSaga for SagaSnapshotCreate {
             ACTION_GENERATE_ID.as_ref(),
         ));
 
+        builder.append(Node::action(
+            "destination_volume_id",
+            "GenerateDestinationVolumeId",
+            ACTION_GENERATE_ID.as_ref(),
+        ));
+
+        // At the beginning of this saga, save disk generation number. Compare
+        // later to see if the disk changed mid-saga.
+        builder.append(Node::action(
+            "disk_gen",
+            "SaveDiskGenToCtx",
+            SAVE_DISK_GEN_TO_CONTEXT.as_ref(),
+        ));
+
+        // Allocate region space for snapshot to store blocks post-scrub
```

Comment:
Would it make sense to share the action implementations between the two sagas that use this (instead of sharing the functions)?  Or maybe even make a subsaga?  (I don't know the answer and this isn't that important.  I'm just thinking out loud as we start sharing more stuff what the most maintainable way will be.  We can always iterate later too.)

Reply:
Some initial thoughts: if we're sharing action implementations that need to pull out of parameters, there's a snag because each saga has it's own Params type, so: 

```rust
async fn generic_alloc_regions(
    sagactx: NexusActionContext,
) -> Result<Vec<(db::model::Dataset, db::model::Region)>, ActionError> {
    let osagactx = sagactx.user_data();
    let params = sagactx.saga_params::<Params>()?;
```

wouldn't work due to Params being different each time. It'd have to be a subsaga with two nodes - one for alloc, one for ensure. Another complication is that each saga will want to do something different for the VolumeConstructionRequest that results from ensuring the regions. This leads to a subsaga that outputs `Vec<(db::model::Dataset, db::model::Region)>`, and is consumed by the parent saga for use in the construction request.

Let's iterate on it later. :)